deploy/template.yaml

```diff
@@ -1247,19 +1297,31 @@ Resources:
   JourneyEngineStepFunction:
     Type: AWS::Serverless::StateMachine
     Properties:
+      Name: !Sub JourneyEngineStepFunction-${AWS::StackName}
       DefinitionUri: journeyEngineStepFunction.asl.json
       DefinitionSubstitutions:
-        IPVProcessJourneyEventFunctionArn: !Ref IPVProcessJourneyEventFunction.Alias
-        CheckExistingIdentityFunctionArn: !Ref CheckExistingIdentityFunction.Alias
-        BuildCriOauthRequestFunctionArn: !Ref BuildCriOauthRequestFunction.Alias
-        BuildClientOauthResponseFunctionArn: !Ref BuildClientOauthResponseFunction.Alias
-        CheckGpg45ScoreFunctionArn: !Ref CheckGpg45ScoreFunction.Alias
-        EvaluateGpg45ScoresFunctionArn: !Ref EvaluateGpg45ScoresFunction.Alias
-        CallTicfCriLambdaArn: !Ref CallTicfCriFunction.Alias
-        CallDcmawAsyncCriLambdaArn: !Ref CallDcmawAsyncCriFunction.Alias
-        StoreIdentityLambdaArn: !Ref StoreIdentityFunction.Alias
-        ResetSessionIdentityFunctionArn: !Ref ResetSessionIdentityFunction.Alias
-        CheckCoiFunctionArn: !Ref CheckCoiFunction.Alias
+        IPVProcessJourneyEventFunctionArn: !Ref IPVProcessJourneyEventFunction.Version
+        CheckExistingIdentityFunctionArn: !Ref CheckExistingIdentityFunction.Version
+        BuildCriOauthRequestFunctionArn: !Ref BuildCriOauthRequestFunction.Version
+        BuildClientOauthResponseFunctionArn: !Ref BuildClientOauthResponseFunction.Version
+        CheckGpg45ScoreFunctionArn: !Ref CheckGpg45ScoreFunction.Version
+        EvaluateGpg45ScoresFunctionArn: !Ref EvaluateGpg45ScoresFunction.Version
+        CallTicfCriLambdaArn: !Ref CallTicfCriFunction.Version
+        CallDcmawAsyncCriLambdaArn: !Ref CallDcmawAsyncCriFunction.Version
+        StoreIdentityLambdaArn: !Ref StoreIdentityFunction.Version
+        ResetSessionIdentityFunctionArn: !Ref ResetSessionIdentityFunction.Version
+        CheckCoiFunctionArn: !Ref CheckCoiFunction.Version
+      AutoPublishAlias: live
+      DeploymentPreference:
+        Type: !Ref StepFunctionsDeploymentPreference
+        Interval: 10
+        Percentage: 20
```

Comment:
has the canary strategy been agreed with core?

Reply:
Not yet tbc

Reply:
Yep, strategy updated, we will deploy canaries to build first